### PR TITLE
Make p2p shuffle submodules private

### PR DESCRIPTION
**Pull request by fjetter**
Builds on https://github.com/dask/distributed/pull/7185

This makes all shuffle p2p modules "private" and exposes only the extensions (They are imported by the scheduler) and the actual graph generator.

With https://github.com/dask/distributed/pull/7180 this should also include the layer

### CODE REVIEW

**Comment by fjetter** [comment]
cc @wence-

**Review by jrbourbeau — COMMENTED** [review]
Nice -- I haven't taken a detailed look, but in general I like the idea of having private modules instead of a lot of private functions (where it makes sense)

File: distributed/shuffle/_shuffle.py [hunk 1]
```diff
@@ -91,15 +93,15 @@ def rearrange_by_column_p2p(
...
-        annotations=None,
+        name: str,
+        column: str,
+        npartitions: int,
+        npartitions_input: int,
+        ignore_index: bool,
+        name_input: str,
+        meta_input: pd.DataFrame,
+        parts_out: list | None = None,
+        annotations: dict | None = None,
```
Comment (fjetter) [inline distributed/shuffle/_shuffle.py:104]: mea culpa. https://github.com/dask/distributed/pull/7180 + https://github.com/dask/distributed/pull/7185 broke CI

File: distributed/shuffle/_shuffle_extension.py [hunk 2]
```diff
@@ -559,7 +559,9 @@ def split_by_worker(
     if not nrows:
         return {}
     # assert len(df) == nrows  # Not true if some outputs aren't wanted
-    t = pa.Table.from_pandas(df)
+    # FIXME: If we do not preserve the index something is corrupting the
+    # bytestream such that it cannot be deserialized anymore
+    t = pa.Table.from_pandas(df, preserve_index=True)
```
Comment (fjetter) [inline distributed/shuffle/_shuffle_extension.py:564]: This is the default. Now it's just explicit

File: distributed/shuffle/_shuffle.py (hunk 1 above)
Comment (fjetter) [reply distributed/shuffle/_shuffle.py:104]: https://github.com/dask/distributed/issues/7196

**Comment by fjetter** [comment]
test_bad_disk is already a known problem. Currently working on it but won't be fixing inthis PR.

Other failures are the websocket failures that happen everywhere right now

**Pull request merged**